### DISCUSSION TITLE
Original firmware file of the Active Marker Deck

### DISCUSSION THREAD

**Discussion by tmartindev** [bitcraze/discussions#949] [Q&A]
Hello,
I know the Active Marker Deck is closed source but is there a file of the original firmware of this deck ?
Thanks for your help.

**Comment by krichardsson** [comment]
No, we have not released the binary as a file. It is flashed to the deck in the factory and the firmware of the deck is usually not changed by the user. 
May I ask what you want to use it for or what problem you want to solve?

**Reply by tmartindev** [reply]
I had to change the firmware so the leds stays always on when the deck has power. I am using it with the Optitrack.

However, I have another deck with the original firmware so finaly I don't think it is a problem as I can dump its flash memory to patch the modified one.

**Reply by krichardsson** [reply]
Ok, thanks for the information!
Good that you can recover the deck.
You should not need to reflash the deck to use an Optitrack system though, set the [`activeMarker.mode`](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/params/#activemarkermode) parameter to 1 (PWM) and the led settings ('front', 'back', 'left', 'right') to control the brightness. I don't have an Optitrack system to test with but I think this should do the trick.
